presto-docs/src/main/sphinx/connector/hive-storage-caching.rst

```diff
@@ -0,0 +1,195 @@
+==============================
+Hive Connector Storage Caching
+==============================
+
+Querying object storage with the :doc:`Hive connector </connector/hive>` is a
+very common use case for Presto. It often involves the transfer of large amounts
+of data. The objects are retrieved from HDFS, or any other supported object
+storage, by multiple workers and processed on these workers. Repeated queries
+with different parameters, or even different queries from different users, often
+access, and therefore transfer, the same objects.
+
+.. contents::
+  :local:
+  :backlinks: none
+
+Benefits
+--------
+
+Enabling caching can result in significant benefits:
+
+**Reduced load on object storage**
+
+Every retrieved and cached object avoids repeated retrieval from the storage in
+subsequent queries. As a result the object storage system does not have to
+provide the object again and again.
+
+For example, if your query accesses 100MB of objects from the storage, the first
+time the query runs 100MB are downloaded and cached. Any following query uses
+these objects. If your users run another 100 queries accessing the same objects,
+your storage system does not have to do any significant work. Without caching it
+has to provide the same objects again and again, resulting in 10GB of total
+storage to serve.
+
+This reduced load on the object storage can also impact the sizing, and
+therefore the cost, of the object storage system.
+
+**Increased query performance**
+
+Caching can provide significant performance benefits, by avoiding the repeated
+network transfers and instead accessing copies of the objects from a local
+cache. Performance gains are more significant if the performance of directly
+accessing the object storage is low compared to accessing the cache.
+
+For example, if you access object storage in a different network, different data
+center or even different cloud-provider region query performance is slow. Adding
+caching using fast, local storage has a significant impact and makes your
+queries much faster.
+
+On the other hand, if you object storage is already running at very high
+performance for I/O and network access, and your local cache storage is at
+similar speeds, or even slower, performance benefits can be minimal.
+
+**Reduced query costs**
+
+A result of the reduced load on the object storage, mentioned earlier, is
+significantly reduced network traffic. Network traffic however is a considerable
+cost factor in an setup, specifically also when hosted in public cloud provider
+systems.
+
+Architecture
+------------
+
+Caching for distributed storage provides a read-through cache. After first
```

Comment:
Default mode in Rubix is `async`. It means that after a while data will be cached asynchronous. Benefit of that is that reads can be merged together and original query doesn't have to wait for data to be cached first.

`read-through` mode of caching works indeed as described here

Reply:
Updated in new PR